test/e2e/upgrade_test.go

```diff
@@ -178,3 +168,30 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
 
 	return err
 }
+
+// deployWindowsWorkloadAndTester tests if the Windows Webserver deployment is available.
+// This is achieved by creating a Job object that continuously curls the webserver every 5 seconds.
+// returns a tearDown func that must be executed to cleanup resources
+func (tc *testContext) deployWindowsWorkloadAndTester() (func(t *testing.T), error) {
+	// create a Windows Webserver deployment
+	deployment, err := tc.deployWindowsWebServer("win-webserver", nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating Windows Webserver deployment for upgrade test")
+	}
+	// create a clusterIP service which can be used to reach the Windows webserver
+	intermediarySVC, err := tc.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating service for deployment %s", deployment.Name)
+	}
+	// create a Job object that continuously curls the webserver every 5 seconds.
+	testerJob, err := tc.createLinuxJob(windowsWorkloadTesterJob, generateLinuxWorkloadTesterCommand(intermediarySVC.Spec.ClusterIP))
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating linux job %s", windowsWorkloadTesterJob)
+	}
+	// return a cleanup func
+	return func(t *testing.T) {
```

Comment:
Does this need to to be (t *testing,T). 

PS: The test should not fail because there was an error in deleting the objects.

Reply:
>Does this need to to be (t *testing,T).

Good catch, It does not.

Reply:
>PS: The test should not fail because there was an error in deleting the objects.

It does not, there is no error handling in the returning func.

Reply:
Fixed in https://github.com/jrvaldes/windows-machine-config-operator/pull/1/commits/5516d32d2b2dc9c99bb1ca5c3a0001b113b027ee